codice/alliance
# CAL-155 Add support for MPEG KLV PES packets that contain optional fields

#### What does this PR do?

Add support for MPEG KLV PES packets that contain optional fields

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?

@jrnorth @bdeining @jlcsmith 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@beyelerb
@kcwire
#### How should this be tested?

1) Build and install Alliance
2) Configure a video stream in 'Alliance Video'.
3) Stream an MPEG-TS file using FFMPEG:

ffmpeg -re -i vid.ts -map 0 -c copy -f mpegts udp://127.0.0.1:50000

4) Search the catalog and confirm that the video chunks were ingested as expected.

#### Any background context you want to provide?

The military spec (MISB ST 1402) states that optional timestamps fields would not be included in the KLV PES packets, which made the packet header a fixed size. When ffmpeg retransmits an MPEG-TS file, it rewrites the PES packets and includes one of the optional timestamps, which is allowed by the MPEG spec. These extra bytes caused the KLV parsing to fail. The bug fix calculates the correct header size, thus allowing the KLV parsing to succeed.

#### What are the relevant tickets?

CAL-155

#### Screenshots (if appropriate)
#### Checklist:
